examples/asr/experimental/speech_to_text_aug.py

```diff
@@ -0,0 +1,243 @@
+                    audio_tar_filepaths=bg_noise_tar_filepaths[i],
+                    orig_sr=orig_sr,
+                )
+
+        self._apply_noise_rir = apply_noise_rir
+
+    def perturb(self, data):
+        prob = self._rng.uniform(0.0, 1.0)
+
+        if prob < self._rir_prob:
+            self._rir_perturber.perturb(data)
+
+        orig_sr = data.orig_sr
+        if orig_sr not in self._fg_noise_perturbers:
+            orig_sr = max(self._fg_noise_perturbers.keys())
+        fg_perturber = self._fg_noise_perturbers[orig_sr]
+
+        orig_sr = data.orig_sr
+        if orig_sr not in self._bg_noise_perturbers:
+            orig_sr = max(self._bg_noise_perturbers.keys())
+        bg_perturber = self._bg_noise_perturbers[orig_sr]
+
+        data_rms = data.rms_db
+        noise = fg_perturber.get_one_noise_sample(data.sample_rate)
+        if self._apply_noise_rir:
+            self._rir_perturber.perturb(noise)
+        fg_perturber.perturb_with_foreground_noise(
+            data, noise, data_rms=data_rms, max_noise_dur=self._max_duration, max_additions=self._max_additions
+        )
+        noise = bg_perturber.get_one_noise_sample(data.sample_rate)
+        bg_perturber.perturb_with_input_noise(data, noise, data_rms=data_rms)
+
+
+class TranscodePerturbation(perturb.Perturbation):
+    def __init__(self, rng=None):
+        """
+        Audio codec augmentation. This implementation uses sox to transcode audio with low rate audio codecs,
+        so users need to make sure that the installed sox version supports the codecs used here (G711 and amr-nb).
+
+        """
+        self._rng = np.random.RandomState() if rng is None else rng
+        self._codecs = ["g711", "amr-nb"]
+
+    def perturb(self, data):
+        att_factor = 0.8
+        max_level = np.max(np.abs(data._samples))
+        norm_factor = att_factor / max_level
+        norm_samples = norm_factor * data._samples
+        orig_f = NamedTemporaryFile(suffix=".wav")
+        sf.write(orig_f.name, norm_samples.transpose(), 16000)
+
+        codec_ind = random.randint(0, len(self._codecs) - 1)
+        if self._codecs[codec_ind] == "amr-nb":
+            transcoded_f = NamedTemporaryFile(suffix="_amr.wav")
+            rates = list(range(0, 8))
+            rate = rates[random.randint(0, len(rates) - 1)]
+            _ = subprocess.check_output(
+                f"sox {orig_f.name} -V0 -C {rate} -t amr-nb - | sox -t amr-nb - -V0 -b 16 -r 16000 {transcoded_f.name}",
+                shell=True,
+            )
+        elif self._codecs[codec_ind] == "g711":
+            transcoded_f = NamedTemporaryFile(suffix="_g711.wav")
+            _ = subprocess.check_output(
+                f"sox {orig_f.name} -V0  -r 8000 -c 1 -e a-law {transcoded_f.name}", shell=True
+            )
+
+        new_data = AudioSegment.from_file(transcoded_f.name, target_sr=16000)
+        data._samples = new_data._samples[0 : data._samples.shape[0]]
+        return
+
+
+def get_augmentor_dict():
```

Comment:
Please do not use hard coded Perturbation dicts like this. Define in Yaml, so that users can provide hydra level overrides for file paths (and these can remain the default, but practically they will fail on user environments cause paths won't match)